config/initializers/constants.rb

```diff
@@ -1,112 +1,225 @@
+  # General limited query/display elements for popup modals
+  MODAL_SEARCH_LIMIT = 5
+  # Comments limited query/display elements for pages
+  COMMENTS_SEARCH_LIMIT = 10
+  # Activity limited query/display elements for pages
+  ACTIVITY_AND_NOTIF_SEARCH_LIMIT = 10
+
+  #=============================================================================
+  # File and data memory size
+  #=============================================================================
+
+  # Max table JSON size in MB
+  TABLE_JSON_MAX_SIZE_MB = 20
+  # Max uploaded file size in MB
+  FILE_MAX_SIZE_MB = 50
+  # Max uploaded user picture avatar size in MB
+  AVATAR_MAX_SIZE_MB = 0.2
+
+  #=============================================================================
+  # Application space
+  #=============================================================================
+
+  # Minimal space needed for organization (in B)
+  MINIMAL_ORGANIZATION_SPACE_TAKEN = 1.megabyte
+  # Additional space of each file is added to its estimated size to account for
+  # DB indexes size etc.
+  ASSET_ESTIMATED_SIZE_FACTOR = 1.1
+
+  #=============================================================================
+  # Format sizes
+  #=============================================================================
+
+  # Picture size formats
+  MEDIUM_PIC_FORMAT = '300x300>'.freeze
+  THUMB_PIC_FORMAT = '100x100>'.freeze
+  ICON_PIC_FORMAT = '40x40>'.freeze
+  ICON_SMALL_PIC_FORMAT = '30x30>'.freeze
+
+  # Hands-on-table number of starting columns and rows
+  HANDSONTABLE_INIT_COLS_CNT = 5
+  HANDSONTABLE_INIT_ROWS_CNT = 5
+
+  # Screen width which is still suitable for sidebar to be shown, otherwise
+  # hidden
+  SCREEN_WIDTH_LARGE = 928
+
+  #=============================================================================
+  # Time
+  #=============================================================================
+
+  # URL expire time, used for presigned file URLs, because outsiders shouldn't
+  # have access to them, but some buffer time is needed for file to be loaded
+  URL_SHORT_EXPIRE_TIME = 30
+  # Same as URL_EXPIRE_TIME, except for cases where the URL migth be used in
+  # another page, and hence the URL mustn't expire by then (e.g. when generating
+  # report and than using same HTML code in PDF, and consequently same file
+  # URL); it expires in exactly one day
+  URL_LONG_EXPIRE_TIME = 86_400
+
+  #=============================================================================
+  # Application colors
+  #
+  # NOTE: Don't use shortened syntax, e.g. #000 for #000000, as some Gems need
+  # full syntax!
+  #=============================================================================
+
+  TAG_COLORS = [
+    '#6C159E',
+    '#159B5E',
+    '#FF4500',
+    '#008B8B',
+    '#757575',
+    '#32CD32',
+    '#FFD700',
+    '#48D1CC',
+    '#15369E',
+    '#FF69B4',
+    '#CD5C5C',
+    '#C9C9C9',
+    '#6495ED',
+    '#DC143C',
+    '#FF8C00',
+    '#C71585',
+    '#000000'
+  ].freeze
```

Comment:
I think that in ruby if you freeze the array it's entities are still mutable you only can't perform the array operations which mutates this array (split, shift!, pop!, <<, ....). You don't need to freeze this array but it makes no harm, so your choice on that.